larpandoracontent/LArHelpers/LArDiscreteProbabilityHelper.cc

```diff
@@ -0,0 +1,129 @@
+/**
+ *  @file   larpandoracontent/LArHelpers/LArDiscreteProbabilityHelper.cc
+ *
+ *  @brief  Implementation of the discrete probability helper helper class.
+ *
+ *  $Log: $
+ */
+
+#include "larpandoracontent/LArHelpers/LArDiscreteProbabilityHelper.h"
+
+namespace lar_content
+{
+
+template <typename T>
+float LArDiscreteProbabilityHelper::CalculateCorrelationCoefficientPValueFromPermutationTest(const T &t1, const T &t2, 
+    std::mt19937 &randomNumberGenerator, const size_t nPermutations)
+{
+    if (1 > nPermutations)
```

Comment:
Out of interest, what's the reason for the ordering inside the if statement? I feel like it is harder to read than how I'd think of it: "Is nPermutations less than one?"

Reply:
ahhhh my understanding, based solely on a conversation with (I think?) @johnmarshall80, that the comparison speed is faster this way round.  
I do totally agree/concede that it hurts one's brain to resolve that comparison mentally.

Reply:
It also used to be done as compilers were stupider/less picky, and if you did `if (name = "Ryan")` they wouldn't complain and would go ahead and assign name to Ryan (i.e. you did `=` not `==`). Now I think they all complain or warn you.

On the plus side, I've seen it called "Yoda conditions", since it sounds like Yoda is saying it...so that is a pretty great.

Reply:
Yes, @CrossR has it right here. I don't think a comparison can be faster in this way (still need to evaluate LHS and RHS and compare), but you can get the compiler to help flag any accidental assignments, rather than e.g. equality checks, if the literal is on the LHS. Of course, const variables also offer this protection.

Reply:
ah, maybe that's what you'd said to me.  Still, there was a reason for the bizarre ordering, at least I did not forget that.